security/signup-controller.php

```diff
@@ -0,0 +1,39 @@
+<?php
+include "db-connection.php";
+
+$conn = openCon();
+
+$firstname = isset($_POST['firstname']) ? $_POST['firstname'] : false;
+$password = isset($_POST['password']) ? $_POST['password'] : false;
+$role = isset($_POST['role']) ? $_POST['role'] : false;
+
+if ($firstname && $password && $role)
+{
+  if ( strcmp($_POST['role'], "admin")==0 ) die("You are not authorized to do this.");
```

Comment:
You have handled the case for admin, but what would happen if the user tries to register as "root"? Or some other role? Have we currently guarded against only a single abuse?